Assets/MixedRealityToolkit/_Core/Devices/UnityInput/UnityDeviceManager.cs

```diff
@@ -0,0 +1,164 @@
+        protected string[] LastDeviceList;
+
+        public override void Enable()
+        {
+            RefreshDevices();
+        }
+
+        public override void Update()
+        {
+            deviceRefreshTimer += Time.unscaledDeltaTime;
+
+            if (deviceRefreshTimer >= DeviceRefreshInterval)
+            {
+                deviceRefreshTimer = 0.0f;
+                RefreshDevices();
+            }
+
+            foreach (KeyValuePair<string, GenericUnityController> controller in ActiveControllers)
+            {
+                controller.Value?.UpdateController();
+            }
+        }
+
+        public override void Disable()
+        {
+            foreach (var genericOpenVRController in ActiveControllers)
+            {
+                if (genericOpenVRController.Value != null)
+                {
+                    InputSystem?.RaiseSourceLost(genericOpenVRController.Value.InputSource, genericOpenVRController.Value);
+                }
+            }
+
+            ActiveControllers.Clear();
+        }
+
+        /// <inheritdoc/>
+        public override IMixedRealityController[] GetActiveControllers()
+        {
+            return ActiveControllers.Values.ToArray<IMixedRealityController>();
+        }
+
+        protected virtual void RefreshDevices()
+        {
+            var joystickNames = Input.GetJoystickNames();
+
+            if (joystickNames.Length <= 0) { return; }
+
+            if (LastDeviceList != null && joystickNames.Length == LastDeviceList.Length)
+            {
+                for (int i = 0; i < LastDeviceList.Length; i++)
+                {
+                    if (joystickNames[i].Equals(LastDeviceList[i])) { continue; }
+
+                    if (ActiveControllers.ContainsKey(LastDeviceList[i]))
+                    {
+                        var controller = GetOrAddController(LastDeviceList[i]);
+
+                        if (controller != null)
+                        {
+                            InputSystem?.RaiseSourceLost(controller.InputSource, controller);
+                        }
+                        ActiveControllers.Remove(LastDeviceList[i]);
+                    }
+                }
+            }
+
+            for (var i = 0; i < joystickNames.Length; i++)
+            {
+                if (string.IsNullOrEmpty(joystickNames[i]))
+                {
+                    continue;
+                }
+
+                if (!ActiveControllers.ContainsKey(joystickNames[i]))
+                {
+                    var controller = GetOrAddController(joystickNames[i]);
+
+                    if (controller != null)
+                    {
+                        InputSystem?.RaiseSourceDetected(controller.InputSource, controller);
+                    }
+                }
+            }
+
+            LastDeviceList = joystickNames;
+        }
+
+        protected virtual GenericUnityController GetOrAddController(string joystickName)
```

Comment:
Have you tested connecting both gamepad and WMR controllers are the same time?
This virtual / override seems to indicate only one can work at a time.
Plus we never add just the UnityDeviceManager and the UnityOpenVRDeviceManager anywhere, unless you have also made changes to the MR manager?  Plus if both are active, are they not going to conflict (I don't think they would, but just asking)

Reply:
Yes, I've tested with both the WMR controllers, and xbox controller connected at the same time. 

No conflicts there.

Reply:
> Plus we never add just the UnityDeviceManager and the UnityOpenVRDeviceManager anywhere, unless you have also made changes to the MR manager?

The `OpenVRDeviceManager` was already defined in the MRManager. I added the `UnityDeviceManager` next to it.